spec/polaris-management-service.yml

```diff
@@ -901,6 +903,58 @@ components:
       required:
         - roleArn
 
+    S3StorageConfigInfo:
+      type: object
+      description: S3 compatible storage configuration info (MinIO, Dell ECS, Netapp StorageGRID, ...)
+      allOf:
+        - $ref: '#/components/schemas/StorageConfigInfo'
+      properties:
+        credsCatalogAndClientStrategy:
+          type: string
+          enum:
+            - VALUE
+            - ENV_VAR_NAME
+          default: ENV_VAR_NAME
+          example: "ACCESS_KEY"
+          description: When you send key VALUE directly via this command, they should apear in logs. By ENV_VAR_NAME without dollar, only a reference will appear in logs, but the value have to be available as environnement variable in the context where Polaris is running
+        credsVendingStrategy:
+          type: string
+          enum:
+            - TOKEN_WITH_ASSUME_ROLE
+            - KEYS_SAME_AS_CATALOG
+            - KEYS_DEDICATED_TO_CLIENT
```

Comment:
How does this work? What identifies a client?

Reply:
I think the term is correct, I was just stuck trying to understand how the service will track which keys are dedicated to which client.

Reply:
Ok.

It is simply one key for catalog itself, then another unique key for any clients whoever they are. I Let client distinction to the _principal/role/privilege_ level. I think it is hard at the class _storage/credential_ level to stick a pair of keys to each different clients.

It is a basic way, when SKIP_CREDENTIAL_SUBSCOPING_INDIRECTION is True and there is not a temporary token, to not divulge internal catalog key and serve a key that can be deactivated for security concerns without breaking catalog itself.

After discussing with MonkeyCanCode here 
 [Prod Deployment credentials](https://github.com/apache/polaris/discussions/405)  the main advantage in this proposal is that you have not to rely on the main credentials provided at the global Polaris service level.

Today if you revoke the Polaris service credentials for AWS, all catalogs with AWS storages are instantly KO.

In this implementation each catalog is independent. It is the same idea about clients keys, to not breaking catalog when clients keys are revoked or rotated fo security reasons.

Reply:
> you have not to rely on the main credentials provided at the global Polaris service level.

I think this is the key point here. I agree that the experience you describe is bad, but I'm not sure that fixing it should be a blocker for s3compat support (or that this is the right fix).

Would you be okay saving this for later, or carving it out into a different PR? In my view relying on the global credentials in production is universally a bad idea, regardless of what STORAGE_TYPE you're using.